sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -0,0 +1,550 @@
+        if async_url:
+            self._async_url = async_url
+
+        location_url = response.headers.get("location")
+        if location_url:
+            self._location_url = location_url
+
+    def get_status(self, pipeline_response):
+        # type: (PipelineResponseType) -> str
+        """Process the latest status update retrieved from an "Operation-Location" header.
+
+        :param azure.core.pipeline.PipelineResponse response: The response to extract the status.
+        :raises: BadResponse if response has no body, or body does not contain status.
+        """
+        response = pipeline_response.http_response
+        if _is_empty(response):
+            raise BadResponse(
+                "The response from long running operation does not contain a body."
+            )
+
+        body = _as_json(response)
+        status = body.get("status")
+        if not status:
+            raise BadResponse("No status found in body")
+        return status
+
+
+class LocationPolling(LongRunningOperation):
+    """Implements a Location polling.
+    """
+
+    def __init__(self):
+        self._location_url = None
+
+    def can_poll(self, pipeline_response):
+        # type: (PipelineResponseType) -> bool
+        """Answer if this polling method could be used.
+        """
+        response = pipeline_response.http_response
+        return "location" in response.headers
+
+    def get_polling_url(self):
+        """Return the polling URL.
+        """
+        return self._location_url
+
+    def get_final_get_url(self, pipeline_response):
+        # type: (PipelineResponseType) -> Optional[str]
+        """If a final GET is needed, returns the URL.
+
+        :rtype: str
+        """
+        return None
+
+    def set_initial_status(self, pipeline_response):
+        # type: (PipelineResponseType) -> str
+        """Process first response after initiating long running operation.
+
+        :param azure.core.pipeline.PipelineResponse response: initial REST call response.
+        """
+        response = pipeline_response.http_response
+
+        self._location_url = response.headers["location"]
+
+        if response.status_code in {200, 201, 202, 204} and self._location_url:
+            return "InProgress"
+        raise OperationFailed("Operation failed or canceled")
+
+    def get_status(self, pipeline_response):
+        # type: (PipelineResponseType) -> str
+        """Process the latest status update retrieved from a 'location' header.
+
+        :param azure.core.pipeline.PipelineResponse response: latest REST call response.
+        :raises: BadResponse if response has no body and not status 202.
+        """
+        response = pipeline_response.http_response
+        if "location" in response.headers:
+            self._location_url = response.headers["location"]
+
+        return "InProgress" if response.status_code == 202 else "Succeeded"
+
+
+class StatusCheckPolling(LongRunningOperation):
```

Comment:
When do we use this class? raw=True? :)

Reply:
provisionningState polling (not raw)

Reply:
That's mostly the fallback to say "if initial status code is 2xx and no polling were deduced, exit right now". So it's not related to "raw", and will be used only if an operation is not necessarly LRO (200 or 202 for instance, there is a network call that is doing that) or if the annotation is incorrect.

Reply:
Gah, Network.